File: api/report_issue.php
<?php

declare(strict_types=1);

require __DIR__ . '/common.php';

$data = normalized_request_data();
$record = append_jsonl('issues.jsonl', [
    'title' => $data['title'] ?? '',
    'city' => $data['city'] ?? '',
    'category' => $data['category'] ?? '',
    'description' => $data['description'] ?? '',
]);

success_response('Sesizarea a fost trimisa prin AJAX.', $record);

File: cgi/report_issue.py
#!/usr/bin/env python3
from lib_cgi import read_form, render_page, run, save_jsonl


def main():
  data = read_form()
  payload = {
    "title": data.get("title"),
    "city": data.get("city"),
    "category": data.get("category"),
    "description": data.get("description")
  }

  save_jsonl("issues.jsonl", payload)

  render_page(
    "Sesizarea a fost trimisa",
    "Sesizarea a fost primita si adaugata in lista de inregistrari.",
    [
      ("Titlu", payload["title"]),
      ("Oras", payload["city"]),
      ("Categorie", payload["category"]),
      ("Descriere", payload["description"])
    ],
    "../date-salvate.php",
    "Vezi datele salvate",
    "../HTML/probleme-locale.html",
    "Revino la probleme"
  )


run(main)
